### DISCUSSION TITLE
Load state on SSR - Nextjs

### DISCUSSION THREAD

**Discussion by PedroJBrito** [facebook/lexical#3765] [Q&A]
Hi,

We have a webapp build with nextjs and we store data that come from an html editor as an html string. We'd like to make the transition to lexical but we're facing some problems loading the state on server side.

- What is our expectation:
  - we have the html string and we want to recreate the editorState that is injected as initialConfig on LexicalComposer, but doing this on server side
- What do we get so far?
  - As DOMParser is not accessible on server side, we're using JSDOM to create a dom from our html string
  - We create the nodes using $generateNodesFromDOM
  - The next step is where the problem comes from - how can we map our nodes that represent our html string to a editor state that can be injected on LexicalComposer?

Any thoughts on this problem will be really appreciated.

**Comment by trueadm** [comment]
Are you expecting to have the Lexical editor hydrate from the HTML in the DOM? As you mention, you need to use `$generateNodesFromDOM` and imperatively setup your initial editor state with that content.

**Reply by PedroJBrito** [reply]
First of all thank you for your answer. Yes, but  `$generateNodesFromDOM` return the nodes that represent my HTML. From those nodes how can I get the initial state to initialConfig?

**Comment by luanmm** [comment]
Hello, fellow contributors! First of all, thank you for the hard work. This library was built with a great perspective, as a "framework".

I have the reverse scenario in mind: I'm using Lexical as content editor in a project that persists the state (as JSON) in database. Then, based on this state, the user can (re-)edit it or view it.

To make it scale for users who just have read access to the content, I want it to be rendered on server-side (without any dependencies from a client-side perspective). But, with the current implementation of `$generateHtmlFromNodes`, based on "window", it seems not to be possible (even using headless editor).

Could we discuss how to fulfill this kind of scenario?

**Comment by jonathanlal** [comment]
^ comment above would be an awesome feature to have
